mlflow/utils/search_utils.py

```diff
@@ -283,21 +278,11 @@ def does_run_match_clause(cls, run, sed):
                                       "not one of '%s" % (comparator, cls.VALID_TAG_COMPARATORS))
             lhs = run.data.tags.get(key, None)
         elif key_type == cls._ATTRIBUTE_IDENTIFIER:
-            if key in cls.VALID_NUMERIC_ATTRIBUTE_KEYS:
-                if comparator not in cls.VALID_NUMERIC_ATTRIBUTE_COMPARATORS:
-                    raise MlflowException("Invalid comparator '{}' not one of "
-                                          "'{}".format(comparator,
-                                                       cls.VALID_NUMERIC_ATTRIBUTE_COMPARATORS))
-                lhs = getattr(run.info, key)
-                value = float(value)
-            elif key in cls.VALID_STRING_ATTRIBUTE_KEYS:
-                if comparator not in cls.VALID_STRING_ATTRIBUTE_COMPARATORS:
-                    raise MlflowException("Invalid comparator '{}' not one of "
-                                          "'{}".format(comparator,
-                                                       cls.VALID_STRING_ATTRIBUTE_COMPARATORS))
-                lhs = getattr(run.info, key)
-            else:
-                raise MlflowException("Invalid attribute key type '{}'".format(key))
+            if comparator not in cls.VALID_STRING_ATTRIBUTE_COMPARATORS:
+                raise MlflowException("Invalid comparator '{}' not one of "
+                                      "'{}".format(comparator,
+                                                   cls.VALID_STRING_ATTRIBUTE_COMPARATORS))
+            lhs = getattr(run.info, key)
```

Comment:
Because each attribute may have subtly different actual return values (e.g., suppose one were an int), could we make sure we have a test which explicitly tries out each one in an end-to-end manner over either the file_store or sql_store?

Reply:
The current 3 searchables have string values. There is coverage for each of these in `test_sqlalchemy_store` and `test_search_utils` will see if something is missing.